godtools/App/Features/Dashboard/Presentation/Tools/ToolsViewModel.swift

```diff
@@ -73,29 +73,34 @@ class ToolsViewModel: ObservableObject {
             .getLanguagePublisher()
             .assign(to: &$appLanguage)
         
-        $appLanguage.eraseToAnyPublisher()
-            .flatMap({ (appLanguage: AppLanguageDomainModel) -> AnyPublisher<(ViewToolsDomainModel, [SpotlightToolListItemDomainModel]), Never> in
-                
-                return Publishers.CombineLatest(
-                    viewToolsUseCase.viewPublisher(appLanguage: appLanguage),
-                    getSpotlightToolsUseCase.getSpotlightToolsPublisher(appLanguage: appLanguage)
-                )
-                .eraseToAnyPublisher()
-            })
-            .receive(on: DispatchQueue.main)
-            .sink { [weak self] (domainModel: ViewToolsDomainModel, spotlightTools: [SpotlightToolListItemDomainModel]) in
-                    
-                self?.favoritingToolBannerMessage = domainModel.interfaceStrings.favoritingToolBannerMessage
-                self?.toolSpotlightTitle = domainModel.interfaceStrings.toolSpotlightTitle
-                self?.toolSpotlightSubtitle = domainModel.interfaceStrings.toolSpotlightSubtitle
-                self?.filterTitle = domainModel.interfaceStrings.filterTitle
-                
-                self?.spotlightTools = spotlightTools
+        Publishers.CombineLatest(
+            $appLanguage.eraseToAnyPublisher(),
+            toolFilterLanguageSelectionPublisher
+        )
+        .flatMap({ (appLanguage: AppLanguageDomainModel, toolFilterLanguage) -> AnyPublisher<(ViewToolsDomainModel, [SpotlightToolListItemDomainModel]), Never> in
+            
+            let selectedLanguageFilter = toolFilterLanguage.language
+            
+            return Publishers.CombineLatest(
```

Comment:
Hey @rachaelblue don't worry about making this change, but I just noticed the CombineLatest within the flatMap and I know we have this similar scenario throughout the app when the app language changes.
I'll need to test this to see what happens when Tools are updated in the RealmDatabase.  I don't think we'd get the update, I'll need to double check.  Shoot I didn't think about this.

Reply:
Yeah good question.  It seems like it should update since it's in the pipeline.  But also, I'm wondering if it might not retain those publishers inside the flatMap when we don't have a reference to them elsewhere.

Reply:
Yeah that could be because there isn't a reference.  Will do some testing today.

Reply:
Hey @rachaelblue just tested this by setting some breakpoints and manually triggering pull to refresh after the page had completed loading everything.  
Once the sync stored data in RealmDatabase the GetToolsRepository publisher fired again which fired the ViewToolsUseCase which then triggered in the ToolsViewModel sink.  So it's working which is good news.  
I also tested commenting out the UseCase references on the ViewModel and publishers were firing. 

In this scenario both the ViewToolsUseCase and GetToolsRepository implementations immediately return a Publishers.CombineLatest where the GetToolsRepository Publishers.CombineLatest triggers on database changes. 

I remember in the scenario that failed to trigger I had the Publishers.CombineLatest in the middle of a flatMap of a UseCase.  I wonder if it had to do something with it.  I'll see if I can test that scenario again.

Reply:
Ok nice!  Thanks for testing this.  Would be curious to see if there are edge cases to watch out for.